File: src/main/java/uk/co/squadlist/web/controllers/DateRange.java
package uk.co.squadlist.web.controllers;

import org.joda.time.LocalDate;

public class DateRange {

    private final LocalDate start;
    private final LocalDate end;

    private final String month;

    private final boolean current;

    public DateRange(LocalDate start, LocalDate end, String month, boolean current) {
        this.start = start;
        this.end = end;
        this.month = month;
        this.current = current;
    }

    public LocalDate getStart() {
        return start;
    }

    public LocalDate getEnd() {
        return end;
    }

    public String getMonth() {
        return month;
    }

    public boolean isCurrent() {
        return current;
    }

    @Override
    public String toString() {
        return "DateRange{" +
                "start=" + start +
                ", end=" + end +
                ", month='" + month + '\'' +
                ", current=" + current +
                '}';
    }

}

File: src/main/kotlin/uk/co/squadlist/web/controllers/DateRange.kt
package uk.co.squadlist.web.controllers

import org.joda.time.Duration
import org.joda.time.LocalDate
import org.joda.time.YearMonth
import uk.co.squadlist.web.views.DateHelper

data class DateRange(val start: LocalDate?, val end: LocalDate?, val month: YearMonth?, val current: Boolean) {

    fun isCurrent(): Boolean = current

    companion object {
        @JvmStatic
        fun from(month: YearMonth?, startDate: LocalDate?, endDate: LocalDate?): DateRange {
            if (month != null) {
                val monthStartDay = month.toLocalDate(1)
                return DateRange(
                    monthStartDay,
                    monthStartDay.plusMonths(1).minusDays(1),
                    month,
                    false
                )

            } else if (startDate != null && endDate != null) {
                val duration = Duration(startDate.toDateTimeAtCurrentTime(), endDate.toDateTimeAtCurrentTime())
                if (duration.standardDays > 0 && duration.standardDays < 50) {
                    return DateRange(
                        startDate,
                        endDate,
                        null,
                        false
                    )
                }
            }

            return DateRange(
                DateHelper.startOfCurrentOutingPeriod().toLocalDate(),
                DateHelper.endOfCurrentOutingPeriod().toLocalDate(),
                null,
                true
            )

        }
    }

}
